### ISSUE TITLE
FR: Add support for Ableton Link

### ISSUE THREAD

**Issue by marcora** [ideoforms/isobar#28]
It would be great if isobar supported syncing of the timeline to Ableton Link clock

**Comment by ideoforms** [comment]
yes this would be cool! I'll scope it out at some point. Have you used any other Python modules that interface with Link? I am keen to keep isobar pure Python and not add any binary dependencies, but I would definitely be open to optional integration with third-party Python wrappers.

A brief search turns up [LinkToPy](https://github.com/bdyetton/LinkToPy) which might be a good option.

**Comment by marcora** [comment]
Yes, that looks like an option but I think it would be easier for the end
user to have it built into isobar, even though that would mean adding a
compiled binary component to it.

**Comment by loparcog** [comment]
Although it might be overkill, would it be viable create a submodule in Python to listen for Ableton Link packets on the network, much like [Carabiner](https://github.com/Deep-Symmetry/carabiner) does as the baseline of LinkToPy? We could keep everything Python-based, but Carabiner says there are some limitations to this

**Comment by ideoforms** [comment]
I have now implemented Link clock support based on [link-python](https://github.com/gonzaloflirt/link-python), and it's working nicely. The remaining challenge is making it installable, as the link-python repo above doesn't have an official PyPi distribution; [this fork](https://github.com/munshkr/link-python) has some infrastructure for cibuildwheel but I'm not sure if the binaries are actually out there.

**Comment by Bubobubobubobubo** [comment]
Hello @ideoforms: check out https://pypi.org/project/LinkPython-extern/. This is what we use to tackle this issue for Sardine. It may be useful to finish that implementation 😄

**Comment by ideoforms** [comment]
@Bubobubobubobubo Fantastic! That is just what I was looking for! Thanks so much, I will do some tests with this now.
